basestar-database-server/src/main/java/io/basestar/database/DatabaseServer.java

```diff
@@ -311,12 +313,13 @@ public Namespace namespace() {
                             } else {
                                 writeDelete(write, schema, key.getId(), before);
                             }
-                            if (storage.eventStrategy(schema) == Storage.EventStrategy.EMIT) {
+                            if (!Action.Result.Type.NO_OP.equals(actionTypes.get(name)) &&
```

Comment:
Can you also skip this preceding block if the result type is NO_OP, will avoid doing unnecessary storage work:

```
if (before == null) {
    assert after != null;
    writeCreate(write, schema, key.getId(), after);
} else if (after != null) {
    writeUpdate(write, schema, key.getId(), before, after);
} else {
    writeDelete(write, schema, key.getId(), before);
}
```